### PR TITLE
feat: Add a javascript.rs file to solve issue #877

### PR DESCRIPTION
**Pull request by Someet-Git**
- In this PR I have created a javascript.rs file which is to extract dependencies.
- some changes in the mod.rs
Basically to solve #877 .

Hey @wsxiaoys can please check this and suggest any changes.

### CODE REVIEW

File: crates/tabby-scheduler/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+tokio = { workspace = true, features = ["full"] }
```
Comment (wsxiaoys) [inline crates/tabby-scheduler/Cargo.toml:9]: is this necessary?

File: crates/tabby-scheduler/src/dataset/deps/javascript.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+use tabby_common::Package;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct NpmPackage {
+    name: String,
+    version: String,
+}
+
+fn extract_npm_deps<'a, I>(packages: I) -> Vec<Package>
```
Comment (wsxiaoys) [inline crates/tabby-scheduler/src/dataset/deps/javascript.rs:12]: this generates all transitive dependencies, not just `direct` dependencies.

That's why I suggest utilizing https://crates.io/crates/package-lock-json-parser/0.1.1 (which provides a util to extract direct dependencies)

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: crates/tabby-scheduler/Cargo.toml (hunk 1 above)
Comment (Someet-Git) [reply crates/tabby-scheduler/Cargo.toml:9]: was giving error while running test

File: crates/tabby-scheduler/src/dataset/deps/javascript.rs (hunk 2 above)
Comment (Someet-Git) [reply crates/tabby-scheduler/src/dataset/deps/javascript.rs:12]: ok let me try this

**Comment by Someet-Git** [comment]
@wsxiaoys can u review the latest commit?

**Comment by wsxiaoys** [comment]
Hi! Thanks for the updates! Have you verified this will only generates direct dependencies, while skip the transitive ones?

One example is let’s say the repository depends on lib A while lib A depends on lib B, we would really prefer only lib A is extracted for the purpose.

**Comment by Someet-Git** [comment]
Hey @wsxiaoys i have done some changes in the javascript.rs file to import the direct deps, and also verified that.

**Comment by Someet-Git** [comment]
Hey @wsxiaoys 
i'm a bit confused here, for example in the  package-lock.json file the for `client/tabby-agent`
there are `dependencies` and `devdependencies`, so we need to extract the name and version of the same?

I checked that it is extracting dependencies of node_module in the package-lock.json file. Is it require?

**Comment by wsxiaoys** [comment]
Hey - thanks for the ping, will give it a try once I got some bandwidth, mostly I wanna make sure this extract only immediate dependencies (which is not obvious by looking from PR)

**Pull request closed without merging**